source/material_model/reactive_fluid_transport.cc

```diff
@@ -0,0 +1,323 @@
+          // leave the current section.
+          base_model = create_material_model<dim>(prm.get("Base model"));
+          if (SimulatorAccess<dim> *sim = dynamic_cast<SimulatorAccess<dim>*>(base_model.get()))
+            sim->initialize_simulator (this->get_simulator());
+
+          if (this->convert_output_to_years() == true)
+            fluid_reaction_time_scale *= year_in_seconds;
+
+          if (this->get_parameters().use_operator_splitting)
+            {
+              AssertThrow(fluid_reaction_time_scale >= this->get_parameters().reaction_time_step,
+                          ExcMessage("The reaction time step " + Utilities::to_string(this->get_parameters().reaction_time_step)
+                                     + " in the operator splitting scheme is too large to compute melting rates! "
+                                     "You have to choose it in such a way that it is smaller than the 'Melting time scale for "
+                                     "operator splitting' chosen in the material model, which is currently "
+                                     + Utilities::to_string(fluid_reaction_time_scale) + "."));
+              AssertThrow(fluid_reaction_time_scale > 0,
+                          ExcMessage("The Fluid reaction time scale for operator splitting must be larger than 0!"));
+            }
+
+          AssertThrow(this->introspection().compositional_name_exists("porosity"),
+                      ExcMessage("Material model Reactive Fluid Transport only "
+                                 "works if there is a compositional field called porosity."));
+
+          AssertThrow(this->introspection().compositional_name_exists("bound_fluid"),
+                      ExcMessage("Material model Reactive Fluid Transport only "
+                                 "works if there is a compositional field called bound_fluid."));
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+
+      // After parsing the parameters for this model, parse parameters related to the base model.
+      base_model->parse_parameters(prm);
+      this->model_dependence = base_model->get_model_dependence();
+    }
+
+
+
+    template <int dim>
+    bool
+    ReactiveFluidTransport<dim>::
+    is_compressible () const
+    {
+      return base_model->is_compressible();
+    }
+
+
+
+    template <int dim>
+    void
+    ReactiveFluidTransport<dim>::
+    melt_fractions (const MaterialModel::MaterialModelInputs<dim> &in,
+                    std::vector<double> &melt_fractions) const
+    {
+      for (unsigned int q=0; q<in.temperature.size(); ++q)
+        {
+          // A fluid-rock reaction model where no reactions occur. The
+          // the melt (fluid) fraction at any point is thus equal to the
+          // porosity, which is determined by the assigned initial porosity,
+          // fluid boundary conditions, and fluid transport through the model.
```

Comment:
I am not sure that this is true. If there is bound water, the total fluid fraction would not equal the porosity. 
The free fluid fraction is equal to the porosity.

Reply:
Good point, I've modified the documentation and code accordingly.